File: leetcode/weekly-286/2217.cpp
class Solution {
public:    
    vector<long long> kthPalindrome(vector<int>& queries, int intLength) {
        int d = (intLength + 1) / 2;
        int r = intLength / 2;
        long long d10 = 1, r10 = 1;
        for (int i = 1; i < d; ++i) {
            d10 *= 10;
        }
        for (int i = 0; i < r; ++i) {
            r10 *= 10;
        }
        vector<long long> ans;
        for (long long i: queries) {
            if (i > 9 * d10) {
                ans.push_back(-1);
            } else {
                long long x = d10 + i - 1;
                long long y = x;
                long long z = 0;
                for (int i = 0; i < d; ++i) {
                    z = z * 10 + y % 10;
                    y /= 10;
                }
                ans.push_back(x * r10 + z % r10);
            }
        }
        return ans;
    }
};

File: leetcode/weekly-286/2217.rb
# @param {Integer[]} queries
# @param {Integer} int_length
# @return {Integer[]}
def kth_palindrome(queries, int_length)
    d = 10 ** (int_length.pred / 2)
    r = int_length / 2
    queries.map do |i|
        if i > 9 * d
            -1
        else
            hi = (d + i - 1)
            lo = hi.to_s[0...r].reverse.to_i(10)
            hi * 10 ** r + lo
        end
    end
end
